### DISCUSSION TITLE
Can you help me understand why there is a discrepancy in SPAN_HEIGHT_BITS and SPAN_MAX_HEIGHT values between `recastnavigation` and `recast4j`?

### DISCUSSION THREAD

**Discussion by ikpil** [ppiastucki/recast4j#173] [Q&A] [answered]
Hello, I am currently comparing the original source code of recastnavigation with the relevant section in recast4j to understand the differences.

`recastnavigation` code:
```
/// Defines the number of bits allocated to rcSpan::smin and rcSpan::smax.
static const int RC_SPAN_HEIGHT_BITS = 13;
/// Defines the maximum value for rcSpan::smin and rcSpan::smax.
static const int RC_SPAN_MAX_HEIGHT = (1 << RC_SPAN_HEIGHT_BITS) - 1;
```

`recast4j` code:
```
/// Defines the number of bits allocated to rcSpan::smin and rcSpan::smax.
public static int SPAN_HEIGHT_BITS = 20;
/// Defines the maximum value for rcSpan::smin and rcSpan::smax.
public static int SPAN_MAX_HEIGHT = (1 << SPAN_HEIGHT_BITS) - 1;
```

The values for SPAN_HEIGHT_BITS and SPAN_MAX_HEIGHT are set differently between these two code snippets.
Why is the value of SPAN_HEIGHT_BITS greater in recast4j compared to recastnavigation?

**Comment by ppiastucki** [comment, marked as answer]
rcSpan struct is defined as follows in recastnavigation:
```
struct rcSpan
{
	unsigned int smin : RC_SPAN_HEIGHT_BITS;
	unsigned int smax : RC_SPAN_HEIGHT_BITS;
	unsigned int area : 6;
	rcSpan* next;
};
```
Which means `smin` occupies 13 bits, `smax` another 13 bits and `area` the remaining 6 bits of the 32-bit word.

These values are expended to full 32-bit integers in recast4j as follows:
```
public class Span {
    public int smin;
    public int smax;
    public int area;
    public Span next;
}
```
Hence the limits are increased, potentially they could be increased to 31 bits instead of 20 if really needed.

**Comment by ikpil** [comment]
thank you
